source/common/http/conn_pool_grid.h

```diff
@@ -185,6 +191,7 @@ class ConnectivityGrid : public ConnectionPool::Instance,
   std::chrono::milliseconds next_attempt_duration_;
   TimeSource& time_source_;
   Http3StatusTracker http3_status_tracker_;
+  OptRef<AlternateProtocolsCache> alternate_protocols_;
```

Comment:
Out of curiosity, why is this optional?

Reply:
Plumbing! Lots of plumbing! :) I mentioned this above, but making this non-optional require a bunch of plumbing and API changes. I'm working on those for a followup.

Reply:
We really need a plumber emoji (and yaks too). This totally SGTM, but capturing that in a comment or TODO here would be nice?